src/NumericalAlgorithms/Interpolation/Actions/InterpolateToTarget.hpp

```diff
@@ -53,6 +60,11 @@ struct InterpolateToTarget {
       Parallel::ConstGlobalCache<Metavariables>& cache,
       const ArrayIndex& array_index, const ActionList /*meta*/,
       const ParallelComponent* const /*meta*/) noexcept {
+    // Exit early if we are on a substep and we don't want to interpolate
+    // substeps
+    if (not InterpolationTargetTag::should_interpolate(box)) {
```

Comment:
Is every InterpolationTargetTag now required to have a should_interpolate function?  (i.e. if it is missing one gets a compiler error?)  Would it be better to make it optional?  Also, since this changes the interface to InterpolationTarget, it should be documented near the top of InterpolationTarget.hpp.

Also, the comment mentions substeps but the name 'should_interpolate' seems to be completely general and has nothing to do with substeps.  Not sure if the comment should be changed or if the function should be renamed. Maybe replace the name 'should_interpolate' with 'should_interpolate_at_this_temporal_id' or 'should_interpolate_on_this_call' or something like that?

Reply:
Only the interpolation target tags that are used in this action need to have the type alias, and I think this action can only be used for the interpolation target tags that have `is_sequential` be false. I'll also make the function optional with a type traits. I'll update the InterpolationTarget documentation for this as well.

I think it's best to change the comment -- the idea is to have a very generic (trigger-like) boolean function to determine whether it's time to interpolate.